### ISSUE TITLE
Generating UV2 removes material from the inspector.

### ISSUE THREAD

**Issue by swarnimarun** [godotengine/godot#16875]
**Godot version:**
Official Godot 3.0

**OS/device including version:**
Windows 10 64bit

**Issue description:**
Generating UV2 for lightmap removes material from the inspector.

**Steps To Reproduce:**
1. Add a ShaderMaterial to a Mesh Instance.
2. Unwrap UV2 for lightmap in Mesh menu
3. Deselect and reselect the mesh instance.
4. Your Material is now gone.!

**Comment by steincodes** [comment]
Still not fixed.?

**Comment by KoBeWi** [comment]
Still happens in 3.2 alpha3

**Comment by KoBeWi** [comment]
Can anyone still reproduce this bug in Godot 3.2.3 or any later release?

If yes, please ensure that an up-to-date Minimal Reproduction Project (MRP) is included in this report (a MRP is a zipped Godot project with the minimal elements necessary to reliably trigger the bug). You can upload ZIP files in an issue comment with a drag and drop.

(can't test on empty project, because unwrap UV expects an ArrayMesh)

**Comment by decepticlown** [comment]
Hmm? No one reported? It's happening even in 3.3. I'll post the project.

**Comment by Calinou** [comment]
Did you add the material on the Mesh resource or on the MeshInstance? In my experience, it's more reliable to add the material "outside" the Mesh resource (i.e. on the MeshInstance's surface properties).

**Comment by decepticlown** [comment]
I applied both and outside where drop down is names Material below software skinning removes it. 

But surface property on mesh itself keeps the material.

**Comment by decepticlown** [comment]
Ok I've figured it out. Outside the material option doesn't ACTUALLY remove the material.

I cleared the material from both surface and Material prop. Then loaded material in material prop. Uv2 unwrap on mesh. It shows the material but I click on other node to refresh the property editor and then click on meshinstance again and to the surprise material is shown in 3D editor but in property it's gone and gives option for new shaders.

Twist: now if I set surface material with something different it still shows the old material. That means in prop editor there's some kind of visual bug.

How do I post the project?
